marklogic-data-hub-spark-connector/spark-test-project/src/main/java/test/StreamTest.java

```diff
@@ -0,0 +1,104 @@
+package test;
+
+import org.apache.spark.SparkConf;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.SQLContext;
+import org.apache.spark.sql.streaming.StreamingQueryException;
+import org.apache.spark.sql.types.StructType;
+import org.apache.spark.streaming.Durations;
+import org.apache.spark.streaming.api.java.JavaStreamingContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+
+public class StreamTest {
+
+    private static Logger logger = LoggerFactory.getLogger(SparkTest.class);
+
+    private static String host = "localhost";
+    private static String username = "pari";
+    private static String password = "password";
+
+    public static void main(String[] args) throws Exception {
+        setConnectionProperties(args);
+
+        logger.info("Creating JavaSparkContext");
+
+        SparkConf sparkConf = new SparkConf()
+            .setAppName("TestAppName")
+            .setMaster("local[1]")
+            .set("spark.driver.allowMultipleContexts", "true");
+
+        JavaStreamingContext streamingContext = new JavaStreamingContext(sparkConf, Durations.seconds(1));
+        JavaSparkContext javaSparkContext = streamingContext.sparkContext();
+
+        try {
+            Dataset<Row> rows = loadRowsFromTestFile(javaSparkContext);
+            rows.join(rows);
+            writeRowsToDataHub(rows);
+        } finally {
+            logger.info("Closing JavaSparkContext");
+            javaSparkContext.close();
+        }
+    }
+
+    private static void setConnectionProperties(String[] args) {
+        if (args.length != 3) {
+            logger.info("Defaulting to host=localhost and username=test-data-hub-operator");
+            username="test-data-hub-operator";
+        } else {
+            host = args[0];
+            username = args[1];
+            password = args[2];
+            logger.info(String.format("Will write to '%s' as user '%s'", host, username));
+        }
+    }
+
+    private static Dataset<Row> loadRowsFromTestFile(JavaSparkContext javaSparkContext) {
+        SQLContext sqlContext = new SQLContext(javaSparkContext);
+        final String filePath = getTestFilePath();
+        logger.info("Loading from file: " + filePath);
+        StructType structType = sqlContext.read().option("header", true).csv(getTestFilePath()).schema();
+        logger.info("Schema of the input file is "+structType);
+        Dataset<org.apache.spark.sql.Row> rows = sqlContext.readStream().schema(structType).format("csv").option("header", true).csv("src/main/resources/data");
```

Comment:
Nice - so the key is to read the header first and determine the schema from it, and then pass the schema in when calling readStream?

Reply:
Yes i believe so.